packages/adapter/lib/adapter/adapter.js

```diff
@@ -8261,73 +8264,70 @@ function Adapter(options) {
-                            try {
-                                subs = JSON.parse(state.val);
-                            } catch {
-                                logger.error(
-                                    `${this.namespaceLog} Cannot parse subscribes for "${this.autoSubscribe[s]}.subscribes"`
-                                );
-                                return;
-                            }
-                            if (!subs[pattern]) {
-                                return;
-                            }
-                            if (subs[pattern][this.namespace] === undefined) {
-                                return;
-                            }
-                            subs[pattern][this.namespace]--;
-                            if (subs[pattern][this.namespace] <= 0) {
-                                delete subs[pattern][this.namespace];
-                            }
+                        let state;
+                        try {
+                            state = await adapterStates.getState(`system.adapter.${autoSub}.subscribes`);
+                        } catch {
+                            // ignore
+                        }
+                        if (!state || !state.val) {
+                            return;
+                        }
+                        let subs;
+                        try {
+                            subs = JSON.parse(state.val);
+                        } catch {
+                            logger.error(`${this.namespaceLog} Cannot parse subscribes for "${autoSub}.subscribes"`);
+                            return;
+                        }
+                        if (!subs[pattern]) {
+                            return;
+                        }
+                        if (subs[pattern][this.namespace] === undefined) {
+                            return;
+                        }
+                        subs[pattern][this.namespace]--;
+                        if (subs[pattern][this.namespace] <= 0) {
+                            delete subs[pattern][this.namespace];
+                        }
 
-                            // if no other subs are there
-                            if (!Object.keys(subs[pattern]).length) {
-                                delete subs[pattern];
-                            }
-                            this.outputCount++;
-                            adapterStates.setState(`system.adapter.${this.autoSubscribe[s]}.subscribes`, subs);
-                        });
+                        // if no other subs are there
+                        if (!Object.keys(subs[pattern]).length) {
+                            delete subs[pattern];
+                        }
+                        this.outputCount++;
+                        adapterStates.setState(`system.adapter.${autoSub}.subscribes`, subs);
                     }
                 }
             }
 
-            const promises = [];
```

Comment:
Can this have performance issues? In fact now runs sequencially ... before was parallel?

Reply:
yes but most likely super minor

Reply:
changed it back to pormises.all